### ISSUE TITLE
Trash bin for Android devices is not working when group folders are used

### ISSUE THREAD

**Issue by ostasevych** [nextcloud/groupfolders#2419]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Open Android Nextcloud client
2. Try to go to Trash bin
3. Observe Error: Cannot open Trash Bin

### Expected behaviour

The deleted files in the trash bin are shown

### Actual behaviour

Seeing Error message: Cannon open Trash Bin

### Android version

13

### Device brand and model

Google Pixel 5A

### Stock or custom OS?

Stock

### Nextcloud android app version

3.25.0 RC2

### Nextcloud server version

26.0.2

### Using a reverse proxy?

No

### Server error logs

```bash
{"reqId":"wpj90yGeyDGMhGf0HHym","level":3,"time":"2023-04-23T19:21:22+00:00","remoteAddr":"100.96.1.17","user":"user","app":"index","method":"GET","url":"/apps/files_trashbin/preview?fileId=103769&file=%2F%D1%81%D0%BF%D0%B8%D1%81%D0%BE%D0%BA_%D0%B2%D1%96%D0%B4%D0%BF%D1%83%D1%81%D1%82%D0%BA%D0%B0_26-04_01_2023.docx.d1671882942&c=bf159e166c56117235875ffbd862fee8","message":"OCA\\GroupFolders\\Trash\\TrashManager::getTrashItemByFileId(): Return value must be of type array, bool returned in file '/var/www/html/nextcloud/apps/groupfolders/lib/Trash/TrashManager.php' line 65","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36","version":"26.0.1.1","exception":{"Exception":"Exception","Message":"OCA\\GroupFolders\\Trash\\TrashManager::getTrashItemByFileId(): Return value must be of type array, bool returned in file '/var/www/html/nextcloud/apps/groupfolders/lib/Trash/TrashManager.php' line 65","Code":0,"Trace":[{"file":"/var/www/html/nextcloud/lib/private/AppFramework/App.php","line":183,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/Route/Router.php","line":315,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/html/nextcloud/lib/base.php","line":1056,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/html/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","Line":169,"Previous":{"Exception":"TypeError","Message":"OCA\\GroupFolders\\Trash\\TrashManager::getTrashItemByFileId(): Return value must be of type array, bool returned","Code":0,"Trace":[{"file":"/var/www/html/nextcloud/apps/groupfolders/lib/Trash/TrashBackend.php","line":356,"function":"getTrashItemByFileId","class":"OCA\\GroupFolders\\Trash\\TrashManager","type":"->"},{"file":"/var/www/html/nextcloud/apps/files_trashbin/lib/Trash/TrashManager.php","line":112,"function":"getTrashNodeById","class":"OCA\\GroupFolders\\Trash\\TrashBackend","type":"->"},{"file":"/var/www/html/nextcloud/apps/files_trashbin/lib/Controller/PreviewController.php","line":100,"function":"getTrashNodeById","class":"OCA\\Files_Trashbin\\Trash\\TrashManager","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":230,"function":"getPreview","class":"OCA\\Files_Trashbin\\Controller\\PreviewController","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":137,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/App.php","line":183,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/Route/Router.php","line":315,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/html/nextcloud/lib/base.php","line":1056,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/html/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/html/nextcloud/apps/groupfolders/lib/Trash/TrashManager.php","Line":65},"CustomMessage":"--"},"id":"6445861f464c6"}
```

**Comment by ostasevych** [comment]
Are any chances to fix it?..

**Comment by XueSheng-GIT** [comment]
I'm on NC 27.0.1 and groupfolders is enabled. Trash bin doesnt't crash on Android.

**Comment by ostasevych** [comment]
> Are any chances to fix it?..

I am on 26.0.4. When I go to the trash bin in my android app I see the message: Error. Cannot open the trash bin!

The journal says:
```
[cron] Error: TypeError: OCA\GroupFolders\ACL\ACLManagerFactory::getACLManager(): Argument #1 ($user) must be of type OCP\IUser, null given, called in /var/www/html/nextcloud/apps/groupfolders/lib/Mount/MountProvider.php on line 206 at <<closure>>

0. /var/www/html/nextcloud/apps/groupfolders/lib/Mount/MountProvider.php line 206
   OCA\GroupFolders\ACL\ACLManagerFactory->getACLManager()
1. /var/www/html/nextcloud/apps/groupfolders/lib/Versions/VersionsBackend.php line 172
   OCA\GroupFolders\Mount\MountProvider->getMount()
2. /var/www/html/nextcloud/apps/groupfolders/lib/Versions/GroupVersionsExpireManager.php line 68
   OCA\GroupFolders\Versions\VersionsBackend->getAllVersionedFiles()
3. /var/www/html/nextcloud/apps/groupfolders/lib/Versions/GroupVersionsExpireManager.php line 59
   OCA\GroupFolders\Versions\GroupVersionsExpireManager->expireFolder()
4. /var/www/html/nextcloud/apps/groupfolders/lib/BackgroundJob/ExpireGroupVersions.php line 43
   OCA\GroupFolders\Versions\GroupVersionsExpireManager->expireAll()
5. /var/www/html/nextcloud/lib/public/BackgroundJob/Job.php line 78
   OCA\GroupFolders\BackgroundJob\ExpireGroupVersions->run()
6. /var/www/html/nextcloud/lib/public/BackgroundJob/TimedJob.php line 103
   OCP\BackgroundJob\Job->start()
7. /var/www/html/nextcloud/lib/public/BackgroundJob/TimedJob.php line 93
   OCP\BackgroundJob\TimedJob->start()
8. /var/www/html/nextcloud/cron.php line 152
   OCP\BackgroundJob\TimedJob->execute()
```

**Comment by ostasevych** [comment]
UPD: after running `occ groupfolders:trashbin:cleanup` command the error disappeared. Well, I feel that is somehow connected with permissions, which should be solved somehow.

**Comment by joshtrichards** [comment]
Does this *only* occur from the Android client or do Trash transactions fail from the Web Files UI as well?

**Comment by ostasevych** [comment]
> Does this _only_ occur from the Android client or do Trash transactions fail from the Web Files UI as well?

Hi! I've tried to study the case myself and found out that this _may_ occur, when I joined some group and left it and some files were deleted. It's been appeared only on mobile client. 

I thing there's need to re-test the incident with the last updates of the client and the groupfolder app.

**Comment by joshtrichards** [comment]
Duplicate of #1998 
Fixed by #2334

Released in GroupFolders v14.0.2 https://github.com/nextcloud/groupfolders/releases/tag/v14.0.2
